src/Agent.Listener/Configuration/AutoLogonManager.cs

```diff
@@ -75,6 +75,34 @@ public override void Initialize(IHostContext hostContext)
                 _terminal.WriteError(StringUtil.Loc("InvalidAutoLogonCredential"));
             }
 
+            _autoLogonRegManager.GetAutoLogonUserDetails(out string currentAutoLogonUserDomainName, out string currentAutoLogonUserName);
+
+            if (currentAutoLogonUserName != null
+                    && !userName.Equals(currentAutoLogonUserName, StringComparison.CurrentCultureIgnoreCase)
+                    && !domainName.Equals(currentAutoLogonUserDomainName, StringComparison.CurrentCultureIgnoreCase))
+            {
+                string currentAutoLogonAccount = String.Format("{0}\\{1}", currentAutoLogonUserDomainName, currentAutoLogonUserName);
+                if (string.IsNullOrEmpty(currentAutoLogonUserDomainName) || currentAutoLogonUserDomainName.Equals(".", StringComparison.CurrentCultureIgnoreCase))
+                {
+                    currentAutoLogonAccount = String.Format("{0}\\{1}", Environment.MachineName, currentAutoLogonUserName);
+                }
+
+                Trace.Warning($"AutoLogon already enabled for {currentAutoLogonAccount}.");    
+                if(!command.GetOverwriteAutoLogonSettings(currentAutoLogonAccount))
+                {
+                    if (command.Unattended)
```

Comment:
regardless unattended or interactive configure, as long as user choose to not overwrite existing setting we should error out.